core/services/workflows/syncer/engine_registry.go

```diff
@@ -0,0 +1,64 @@
+package syncer
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/workflows"
+)
+
+type engineRegistry struct {
+	engines map[string]*workflows.Engine
+	mu      sync.RWMutex
+}
+
+func newEngineRegistry() *engineRegistry {
+	return &engineRegistry{
+		engines: make(map[string]*workflows.Engine),
+	}
+}
+
+// Add adds an engine to the registry.
+func (r *engineRegistry) Add(id string, engine *workflows.Engine) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.engines[id] = engine
+}
+
+// Get retrieves an engine from the registry.
+func (r *engineRegistry) Get(id string) *workflows.Engine {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.engines[id]
+}
+
+// Remove removes an engine from the registry.
+func (r *engineRegistry) Remove(id string) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	engine, ok := r.engines[id]
+	if !ok {
+		return errors.New("remove failed: engine not found")
+	}
+	err := engine.Close()
```

Comment:
This makes the API slightly asymmetric, since Remove will close the engine but Add doesn't start it;

In general I would prefer to leave closing the engine to the business logic; it's more obvious to read a handler that:
- pops something from the registry
- closes it
- deletes a record in the database
- ...

Then one that does:
- removes an item from the registry
    - implicitly closes it
- remove a database record

It also keeps all the side effects on the same abstraction level